hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/WALEntryBatch.java

```diff
@@ -52,6 +52,9 @@ class WALEntryBatch {
   private Map<String, Long> lastSeqIds = new HashMap<>();
   // indicate that this is the end of the current file
   private boolean endOfFile;
+  // indicate the buffer size used, which is added to
+  // ReplicationSourceWALReader.totalBufferUsed
+  private long usedBufferSize;
```

Comment:
Since we have recorded the size here, I think we could use it directly in many places(especially in ReplicationSourceShipper), so we do not need to calculate the size of the WALEntryBatch again?

Anyway, can be a follow on issue.

Reply:
@Apache9 ,yes,  I plan to open a new PR to centralize the totalBufferUsed related code and eliminate duplicate code